Cell Battery Multiplier

rocketguy2 opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Please fill out this form, so that we have all the info needed to review your changes -->
<!-- Any text inside of the angle brackets will not show up in the final comment. Check the Preview tab to confirm -->

## Description
<!-- Describe your changes in detail -->
<!-- If you are adding new content, please let us know if it is based on some canon, and provide a reference to it -->

This is a change to the Cell Battery Oak Item, which replaces the current effect of decreasing the number of charges needed to activate the discharge, to multiplying the amount of charge gained per underground action. In addition, this also adds a new farm aura to the Kee Berry, which also multiplies the Underground Battery charge. This also allows for the underground charge to showcase a charge value up to one decimal place.

## Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- Please also link to any related issues or PRs here. -->
<!-- You can link an issue to be auto-closed when this is merged by writing 'Closes #1234' or 'Fixes #1234' -->

This is an implementation of, and therefore Closes #6063 and as such, my motivation for doing this is explained there. Adding the effect to the Kee Berry was to give an example of what effects making battery charge a multiplier could have outside of modifying the Cell Battery. The Kee Berry was selected due to its current lack of purpose, but I believe this effect would be better off on a berry with a shorter lifespan, with the Kee Berry being the shortest lasting berry Gen IV+ that does not currently have an effect.

## How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

This was tested by compiling the code as is, and checking whether or not equipping the Cell Battery would cause the underground charge to increase at an increased rate. When it did, I further modified the code to make it so that at the max level, the Cell Battery would work on a 1.5x multiplier, to make sure the intermediate levels would work properly. When I got that working, I set the code back to how it is now. Some minor text edits were required to ensure that no graphical errors occur, removing the spaces in the meter showing the underground charge in the underground screen.

In order to test the new Farm Aura, I applied it to the Cheri Berry, in order to make the test easier. When I was able to see that it worked, I moved the aura to the Kee Berry afterwards.

## Types of changes
<!-- What types of changes does your code introduce? Delete any that don't apply, and/or add more you think would be useful -->

- New feature
- Modified pre-existing feature

rocketguy2 committed:
Implement PR#6063

This is an implementation of PR#6063, in which the Cell Battery Oak Item is modified such that, instead of decreasing the maximum number of charges needed to activate the Cell Battery, it increases the number of charges added to the Cell Battery whenever it would usually add one.

rocketguy2 committed:
Adding Battery Charge multiplier to Kee Berry

This adds the Battery Charge aura to the farm, which has the same effect as the new implementation of the Cell Battery, increasing the amount of charge gained per underground action. In addition, it applies this effect to the Kee Berry.

rocketguy2 committed:
replaced batteryCharge with undergroundCharge

RegisCoaxans committed:
Merge branch 'develop' into CellBatteryMultiplyTest

RegisCoaxans reviewed (changes requested):
The aura feels out of place, it would need to be way too OP to be relevant, yet I don't think anyone is going to go for a faster discharge setup in the farm. Please revert this change.
Also, since this makes the needed charges unvarying, we don't need to show `/ 60`; either it could be removed or the current charges could display in percentage of needed charges.

rocketguy2 committed:
Reverted Farm Aura Additions

Reverted the earlier commit that introduced the farm aura for Underground Charge and its application to the Kee Berry.

rocketguy2 committed:
Updated Underground Charge Counter to Percentage

Updated the counter used to track the progress of the Underground Charge from a number into a percentage, same as the rest of the underground tools. Additionally ensured that the percentage cannot exceed 100%.

rocketguy2 committed:
Update underground.html

For some reason this change didn't pass through to the actual commit, whoops.

RegisCoaxans committed:
Less magic numbers

RegisCoaxans committed:
Less magic numbers

RegisCoaxans committed:
Merge branch 'develop' into CellBatteryMultiplyTest
